sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -164,6 +156,18 @@ public boolean start() throws IOException {
     // since CPU consumed on the workers would be low and will likely avoid unnecessary upscale.
     offsetConsumerConfig.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, "read_uncommitted");
 
+    Object groupId = spec.getConsumerConfig().get(ConsumerConfig.GROUP_ID_CONFIG);
+    // override group_id and disable auto_commit so that it does not interfere with main consumer
+    String offsetGroupId =
+        String.format(
+            "%s_offset_consumer_%d_%s",
+            name, (new Random()).nextInt(Integer.MAX_VALUE), (groupId == null ? "none" : groupId));
+    offsetConsumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, offsetGroupId);
+
+    if (spec.getOffsetConsumerConfig() != null) {
```

Comment:
Another side effect is that this could overwrite 'ISOLATION_LEVEL_CONFIG' too, if the user's config has that (intentionally or not). May be we don't want that. 

I think it will be simpler to move this if block to after line 152 (line number on master).

Reply:
+1